torchmultimodal/utils/attention.py

```diff
@@ -43,6 +45,32 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
     return extended_attention_mask
 
 
+def get_extended_attention_mask_for_decoder(attention_mask: Tensor) -> Tensor:
```

Comment:
I'm still not 100% sure why we need this. Why not just call get_causal_attention_mask and get_extended_attention_mask(attn_mask) then take the product in the decoder itself?